lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1908,6 +1908,89 @@ namespace Js
         return SubString::New(pThis, idxStart, span);
     }
 
+    Var JavascriptString::EntryPadStart(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        Assert(!(callInfo.Flags & CallFlags_New));
+        CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(PadStartCount);
+
+        JavascriptString * pThis = nullptr;
+        GetThisStringArgument(args, scriptContext, L"String.prototype.padStart", &pThis);
+
+        return PadCore(args, pThis, true /*isPadStart*/, scriptContext);
+    }
+
+    Var JavascriptString::EntryPadEnd(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        Assert(!(callInfo.Flags & CallFlags_New));
+        CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(PadEndCount);
+
+        JavascriptString * pThis = nullptr;
+        GetThisStringArgument(args, scriptContext, L"String.prototype.padEnd", &pThis);
+
+        return PadCore(args, pThis, false /*isPadStart*/, scriptContext);
+    }
+
+    JavascriptString* JavascriptString::PadCore(ArgumentReader& args, JavascriptString *mainString, bool isPadStart, ScriptContext* scriptContext)
+    {
+        Assert(mainString != nullptr);
+        Assert(args.Info.Count > 0);
+
+        if (args.Info.Count == 1)
+        {
+            return mainString;
+        }
+
+        int64 maxLength = JavascriptConversion::ToLength(args[1], scriptContext);
+        charcount_t currentLength = mainString->GetLength();
+        if (maxLength <= currentLength)
+        {
+            return mainString;
+        }
+
+        JavascriptString * fillerString = nullptr;
+        if (args.Info.Count > 2 && !JavascriptOperators::IsUndefinedObject(args[2], scriptContext))
+        {
+            JavascriptString *argStr = JavascriptConversion::ToString(args[2], scriptContext);
+            if (argStr->GetLength() > 0)
+            {
+                fillerString = argStr;
+            }
+        }
+
+        if (fillerString == nullptr)
+        {
+            fillerString = NewWithBuffer(L" ", 1, scriptContext);
+        }
+
+        Assert(fillerString->GetLength() > 0);
+
+        charcount_t fillLength = (charcount_t)(maxLength - currentLength); // Possible truncation.
```

Comment:
@Cellule  Our string does not go more than maxcharcount. 
so in this case there could be one of the two possibilities. 
a) Throw OOM or b) wrap around.
I chose to do wrap around earlier, but now that you pointed that out I looked at other API (eg. repeat) and it throws in such cases. 
I going to match with that and will throw on overflow condition.